src/hardware.h

```diff
@@ -56,4 +55,27 @@ void start_stm32_bootloader();
  */
 void reset_device();
 
+/**
+ * Implements watchdog for multiple threads in software
+ */
+void sw_watchdog(struct k_timer *timer_id);
+
+/**
+ * Configure timer for software watchdog and
+ * configure actual watchdog
+ */
+void start_sw_watchdog();
```

Comment:
Can you rename to `sw_watchdog_start` (more consistent with other function names).